src/components/Button/Button.js

```diff
@@ -61,31 +64,35 @@ const Button = ({
     return null;
   })();
 
-  const button = (
-    <button
-      {...other}
-      {...commonProps}
-      disabled={disabled}
-      type={type}
-      ref={other.inputref}>
-      {children}
-      {buttonImage}
-    </button>
-  );
-
-  const anchor = (
-    <a
-      {...other}
-      {...commonProps}
-      href={href}
-      role="button"
-      ref={other.inputref}>
-      {children}
-      {buttonImage}
-    </a>
+  let component = 'button';
+  let otherProps = {
+    disabled,
+    type,
+  };
+  const anchorProps = {
+    role: 'button',
+    href,
+  };
+  if (as) {
+    component = as;
+    otherProps = {
+      ...otherProps,
+      ...anchorProps,
+    };
+  } else if (href) {
+    component = 'a';
+    otherProps = anchorProps;
+  }
+  return React.createElement(
```

Comment:
i think the jsx form might be clearer but either way works

Reply:
it seems that the main difference is `createElement` automatically forwards refs, but we get an `inputRef` instead of a `ref` anyway so that part isn't strictly relevant